### PR TITLE
CGD-204: Did-Document Resolver Component

### PR DESCRIPTION
**Pull request by bjoern-arnold**
* Interface DidResolver deprecates DidDocumentResolver
* Added implementations / tests
  * DidWebResolver
  * DidUniResolver (adapter for [DIF Universal Resolver](https://github.com/decentralized-identity/universal-resolver/))
  * CompositeDidResolver
* Extended documentation

### CODE REVIEW

File: cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolver.java [hunk 1]
```diff
@@ -23,6 +23,10 @@
 import org.eclipse.tractusx.ssi.lib.model.did.DidDocument;
 import org.eclipse.tractusx.ssi.lib.model.did.DidMethod;
 
+@Deprecated
+/**
+ * @deprecated replaced by {@link DidResolver}
+ */
 public interface DidDocumentResolver {
```
Comment (DominikPinsel) [inline cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolver.java:30]: I assume we can deprecate the DidDocumentResolverRegistry, too?

File: cx-ssi-lib/src/test/java/org/eclipse/tractusx/ssi/lib/did/web/DidWebResolverIT.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    assertEquals(new DidDocument(TestResourceUtil.getPublishedDidDocument()), actualDidDoc);
+  }
+
+  @Test
+  public void shouldResolveValidExternalWebDid() throws DidResolverException {
+    final String didIdentifier = "did.actor:alice";
+    Did validDidWeb = new Did(new DidMethod("web"), new DidMethodIdentifier(didIdentifier));
+    assertTrue(httpsResolver.isResolvable(validDidWeb));
+    DidDocument actualDidDoc = httpsResolver.resolve(validDidWeb);
+    assertEquals("did:web:" + didIdentifier, actualDidDoc.get("id"));
```
Comment (DominikPinsel) [inline cx-ssi-lib/src/test/java/org/eclipse/tractusx/ssi/lib/did/web/DidWebResolverIT.java:77]: Probably I'm just confused. If we resolve the variables in our head, doesn't  the assertEquals compare this?
`assertEquals("did:web:did.actor:alice", "did:web:localhost%3A8081");`

**Review by DominikPinsel — COMMENTED** [review]
Thank you this this PR. Also good catch in deprecating the DidDocumentRegistry thing, the composite pattern here is really a lot better :)

File: cx-ssi-lib/src/test/java/org/eclipse/tractusx/ssi/lib/did/web/DidWebResolverIT.java (hunk 2 above)
Comment (bjoern-arnold) [reply cx-ssi-lib/src/test/java/org/eclipse/tractusx/ssi/lib/did/web/DidWebResolverIT.java:77]: There are two tests:
- shouldResolveValidWebDid resolves a DID on localhost (hosted on nginx service started by testcontainers)
- shouldResolveValidExternalWebDid resolves the (publicly hosted) did "did:web:did.actor:alice" from https://did.actor/alice/did.json

File: cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolver.java (hunk 1 above)
Comment (bjoern-arnold) [reply cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolver.java:30]: Yes, thanks for the hint.

File: cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidUniResolver.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+   * Always returns {@code true} since there is no easy way to determine which drivers are present
+   * in the universal resolver.
+   *
+   * @param did a valid Decentralized Identifier (according to <a
+   *     href="https://www.w3.org/TR/did-core/">W3C DID specification</a>)
+   * @return <em>true</em>
+   */
+  @Override
+  public boolean isResolvable(Did did) {
+    return true;
```
[the remark this answers is not in the document]
Comment (bjoern-arnold) [reply cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidUniResolver.java:93]: Good point. I see the universal resolver as a fallback that should be the last in the resolution chain. With that configuration it doesn't "hurt" (i.e. prevent another, more specific resolver to do the job) if the UniResolver always tries to resolve the DID. Also, the result may not only be evaluated based on the DID method but also on the connected VDRs, etc.
What do you think?

File: cx-ssi-lib/src/test/java/org/eclipse/tractusx/ssi/lib/did/web/DidWebResolverIT.java (hunk 2 above)
Comment (DominikPinsel) [reply cx-ssi-lib/src/test/java/org/eclipse/tractusx/ssi/lib/did/web/DidWebResolverIT.java:77]: Ah,now I see it. I did not know that `did.actor:alice` can be resolved to a real DID :)

File: cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidUniResolver.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply cx-ssi-lib/src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidUniResolver.java:93]: Yea, this is true. Then let's just keep it as it is :)

**Review by DominikPinsel — APPROVED** [review]
Great PR :)

**Pull request merged**